### ISSUE TITLE
NotificationStore DeleteNotificationAsync not setting TenantId

### ISSUE THREAD

**Issue by jmhinnen** [aspnetboilerplate/module-zero#278]
So, currently the code inside the NotificationStore for DeleteNotificationAsync is not setting the tenantId to Null before calling delete.  

Keep in mind, I am using separate databases for host / tenant.

There are two scenarios to send out Notifications, if there are 5 or more users to be notified, it sends it out via a background job (NotificationDistrubutorJob).  If there are less than 5, it notifies the users right then.  In the background job, the delete works because there is no unit of work and I am assuming the context defaults to the Host and deletes it correctly.  When the notifications run in real time, there is a UnitOfWork and the TenantId is set to a value.  When it goes to delete the Notification, it hits the TenantDB and not the HostDB since the UOW is not set to null.  In all the other methods within that class, you are setting the TenantId to Null.  I assume this was just an over site.

So I think NotificationStore DeleteNotificationAsync should be like InsertNotificationAsync:

````C#
[UnitOfWork]
public virtual Task DeleteNotificationAsync(NotificationInfo notification)
{
    using (_unitOfWorkManager.Current.SetTenantId(null))
    {
        await _notificationRepository.Delete(notification);
        await _unitOfWorkManager.Current.SaveChangesAsync();
    }
}
````

Thoughts?

**Comment by hikalkan** [comment]
Hi,

NotificationInfo entity is marked as HOST side. https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/src/Abp/Notifications/NotificationInfo.cs#L15

ABP automatically switches to host when you are working with host side entities. This is why it's not need to set host explicitly (Probably, some of other methods have unnecessary code, I should check it).

Do you have any problem with that? I mean "are notifications being sent correctly?".

**Comment by jmhinnen** [comment]
Let me test it one more time.  Yes, originally we were having an issue with this but for a while there was a bug in the SetCurrentTenant that was not setting the tenant back correctly.  When that was happening, we had to override the method to set it to Null.  So I could be working against another issue that was fixed so this wasn't needed.

To be honest, I think there is an issue, but let me double check.  I have a scenario I can test all setup so I will get back with you today.
